### ISSUE TITLE
Update-DbaServiceAccount fails for SSRS 2019

### ISSUE THREAD

**Issue by d-watkins** [dataplat/dbatools#8635]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

Using the following command to change the domain account for the SSRS 2019 Service:

Get-DbaService ServerName -Type SSRS | Update-DbaServiceAccount -Username 'Domain\User'

Returns the following error:

writeErrorStream      : True
PSMessageDetails      :
Exception             : System.Exception: You cannot call a method on a null-valued expression.
TargetObject          :
CategoryInfo          : NotSpecified: (:) [Write-Error], Exception
FullyQualifiedErrorId : dbatools_Update-DbaServiceAccount,Stop-Function
ErrorDetails          : You cannot call a method on a null-valued expression.
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, D:\git\dbatools\allcommands.ps1: line 149023
                        at Update-DbaServiceAccount<End>, D:\git\dbatools\allcommands.ps1: line 136274
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}

The same command succeeds for types engine and agent

### Steps to Reproduce

```powershell
# provide your command(s) executed pertaining to dbatools
# please include variable values (redacted or fake if needed) for reference
```
Get-DbaService ServerName -Type SSRS | Update-DbaServiceAccount -Username 'Domain\User'

### Please confirm that you are running the most recent version of dbatools

Running version 1.1.140

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe), Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version

PSVersion 5.1.14393.5127

### SQL Server Edition and Build number

SQL 2019 RTM-CU18

### .NET Framework Version

4.6.81

**Comment by mattcargile** [comment]
So my guess, is your system hasn't fully initialized SSRS. I just looked at a fresh install of SSRS 2019 and it doesn't exist in `New-Object Microsoft.SqlServer.Management.Smo.Wmi.ManagedComputer` `Services` properties which is the object that is used to change the password. 

Additionally, previous full installs I have for 2017 and 2016 show a different `ServiceName` and `InstanceName`. For my two 2019 SSRS instances, they show `SQLServerReportingServices` for the `ServiceName` and `SSRS` for the `InstanceName`. My fully configured instances have the `InstanceName` actually defined as what I configured it to be or `MSSQLSERVER`.

Can you return the output of `Get-DbaService ServerName -Type SSRS | Format-Text -AutoSize`? My guess is you need to configure the server through the GUI or run `rsconfig`.

**Comment by mattcargile** [comment]
I was off-base on my previous comment. Now I just think `ManagedComputer` can no longer be used to manage _SSRS 2019_. It appears [this](https://learn.microsoft.com/en-us/sql/reporting-services/wmi-provider-library-reference/configurationsetting-method-setwindowsserviceidentity?view=sql-server-ver16) is new method to be used?

**Comment by andreasjordan** [comment]
@nvarscar / @potatoqualitee - Do we want to support Reporting Services with this command?

**Comment by nvarscar** [comment]
if we already have required libraries loaded in the module - I don't see why not. Otherwise, maybe this might be out of scope for dbatools.

**Comment by wsmelton** [comment]
Since SSRS is a separate download and no longer a component included with SQL Server I would consider it out of scope.

**Comment by mattcargile** [comment]
Yeah I can see both viewpoints if supporting and not supporting it. Maybe there could be a warning that SSRS isn't supported?

**Comment by andreasjordan** [comment]
Ok, opened PR with documentation changes.

**Comment by mattcargile** [comment]
Couldn't there be a check in the code to block any SSRS version after 2016. Or just block it altogether? The error isn't all that clear for the user.

**Comment by mattcargile** [comment]
> We just don't have information about the version. Or am I missing something?

We have the `Type` being `SSRS`, right?

**Comment by mattcargile** [comment]
> So we should block SSRS completly? Because we can not block "SSRS after version 2016".

Yeah that was kind of my question. We can look inside the Managed Computer object ( e.g. `[Microsoft.SqlServer.Management.Smo.Wmi.ManagedComputer]::new('server').services` and if it can't see the SSRS then it can throw a message like it can't find it and SSRS may not be supported instead of the more confusing null error.

**Comment by nvarscar** [comment]
How would you feel about just adding a Warning that would say that SSRS beyond 2016 is not supported by this command any time there is an SSRS service in the pipe? It would then fail on its own, but at least give a warning about "why". I know it might be noisy for previous versions of SSRS, but they might be just old enough so that the impact wouldn't be as widespread.

**Comment by mattcargile** [comment]
> How would you feel about just adding a Warning that would say that SSRS beyond 2016 is not supported by this command any time there is an SSRS service in the pipe? It would then fail on its own, but at least give a warning about "why". I know it might be noisy for previous versions of SSRS, but they might be just old enough so that the impact wouldn't be as widespread.

That would be the minimum I would be ok with because from the user experience, the null exception isn't helpful. 

Ideally, since we already are querying the `Services` property on `[Microsoft.SqlServer.Management.Smo.Wmi.ManagedComputer]::new()` , I would much _much_ rather see a more descriptive error message saying it can't find the Service and the SSRS version is incompatible and maybe point to some of the new docs on and another library.

**Comment by andreasjordan** [comment]
The problem is: You are talking about the remote code. So more changes are needed to get the information to the local side and then react to it. Will try to find time for that...

**Comment by andreasjordan** [comment]
We have more problems: As we use `Invoke-ManagedComputerCommand`, we can not catch all the exceptions as they are used in that command in a big try-catch-tree. I need more time for this...
